text/docs-code-contributions.md

```diff
@@ -0,0 +1,59 @@
+- Start Date: 2019-02-07)
+- RFC PR: 
+- Gatsby Issue:
+
+# Summary
+
+Reorganize the contributor documentation to aid in Gatsby core and docs development.
+
+# Motivation
+
+The Gatsby.js docs, while wonderful, are light on coding instructions for contributors. With dedicated sections in the Contributing docs, it will be easier to learn how to clone the repository, where to find certain things like docs or plugins, how to run tests, and insider infrastructure knowledge required to submit a successful pull request on the first try.
+
+A lot of this information already lives on the "How to Contribute" page, but it's buried. It would help discoverability if this information lived on separate pages.
+
+# Detailed design
+
+I propose that we add some pages to the existing contributor documentation and move some content around. Here is an IA diagram that (briefly) reflects the current structure and what I think we should add (open for discussion, of course):
+
+Contributing
+- Community
+    - Pair Programming Sessions
+    - Where to participate in the community (this was in RFCs, perhaps it could be reintroduced and content moved from "How to contribute"?)
+    - How to run a Gatsby workshop*
+    - How to pitch Gatsby*
+- Code of Conduct
+- Gatsby Style Guide
+- How to contribute
+    - How to file an issue
+    - How to label an issue
+    - Free swag for contributors
+    - Contributing to the documentation
+        - Docs templates
+    - Contributing to the repo
+        - page includes moved content from "How to contribute" incl. starter library, website, everything that isn't docs (with a mention of plugins section).
+    - Setting up test environments (Wordpress, Contentful, etc.)
+- RFC process
+- Submit to Site Showcase
+- Submit to Creator Showcase
+- Submit to Starter Library
+- Submit to Plugin Library
+
+# Drawbacks
+
+Why should we *not* do this? Please consider:
+
+- we will need some redirects so people who have bookmarked things don't get 404s
```

Comment:
We should do this, but we've not been consistent with updating URLs when rearranging the nav in the past, so it's not the end of the world if we _don't_ do this.

Reply:
@m-allanson I think you could be right--is there a way to know how many people bookmark things? Just out of curiosity.

Reply:
I don't think so, bookmarks are local to a person's browser.